crates/sui/src/keytool.rs

```diff
@@ -878,53 +923,80 @@ impl KeyToolCommand {
 
             KeyToolCommand::ZkLoginSigVerify {
                 sig,
-                tx_bytes,
-                provider,
+                bytes,
+                intent_scope,
                 curr_epoch,
+                network,
             } => {
                 match GenericSignature::from_bytes(
                     &Base64::decode(&sig).map_err(|e| anyhow!("Invalid base64 sig: {:?}", e))?,
                 )? {
                     GenericSignature::ZkLoginAuthenticator(zk) => {
-                        if tx_bytes.is_none() || provider.is_none() || curr_epoch.is_none() {
+                        if bytes.is_none() || curr_epoch.is_none() {
                             return Ok(CommandOutput::ZkLoginSigVerify(ZkLoginSigVerifyResponse {
-                                tx_data: None,
+                                data: None,
                                 parsed: Some(serde_json::to_string(&zk)?),
                                 res: None,
                                 jwks: None,
                             }));
                         }
 
-                        let tx_data: TransactionData = bcs::from_bytes(
-                            &Base64::decode(&tx_bytes.unwrap())
-                                .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
-                        )?;
                         let client = reqwest::Client::new();
-                        let jwks = fetch_jwks(
-                            &OIDCProvider::from_str(&provider.clone().unwrap())
-                                .map_err(|_| anyhow!("Unsupported provider:"))?,
-                            &client,
-                        )
-                        .await?;
+                        let mut jwks = Vec::new();
```

Comment:
This seems like a hack. The `iss` info is available in the signature, so we don't have to fetch JWKs for everyone, no?

Reply:
revised